### PR TITLE
Support status codes in `retrieve_latest`

### PR DESCRIPTION
**Pull request by haakonvt**
### Added
- Datapoints method `retrieve_latest` now supports status codes.

### CODE REVIEW

**Review by doctrino — COMMENTED** [review]
Missing a test that checks that `.retrieve_latest` varies depending on whether you `ignore_bad_datapoints` and `treat_uncertain_as_bad`

File: cognite/client/_api/datapoints.py [hunk 1]
```diff
@@ -1271,23 +1277,36 @@ def retrieve_latest(
...
+                ...     "foo",
+                ...     LatestDatapointQuery(external_id="abc", before="3h-ago", target_unit_system="Imperial"),
+                ...     LatestDatapointQuery(external_id="def", include_status=True),
+                ...     LatestDatapointQuery(external_id="ghi", include_status=True, ignore_bad_datapoints=False)]
                 >>> res = client.time_series.data.retrieve_latest(
-                ...     id=id_queries,
-                ...     external_id=LatestDatapointQuery(
-                ...         external_id="abc", before="3h-ago", target_unit_system="Imperial")
-                ... )
+                ...     id=id_queries, external_id=xid_queries)
```
Comment (haakonvt) [inline cognite/client/_api/datapoints.py:1297]: Note to self: add examples on using status codes

File: cognite/client/_api/datapoints.py [hunk 2]
```diff
@@ -1271,23 +1277,37 @@ def retrieve_latest(
...
+            for individual customisation of 'target_unit', 'target_unit_system', 'include_status', 'ignore_bad_datapoints'
+            and 'treat_uncertain_as_bad'.
 
                 >>> from datetime import datetime, timezone
                 >>> id_queries = [
                 ...     123,
                 ...     LatestDatapointQuery(id=456, before="1w-ago"),
                 ...     LatestDatapointQuery(id=789, before=datetime(2018,1,1, tzinfo=timezone.utc)),
                 ...     LatestDatapointQuery(id=987, target_unit="temperature:deg_f")]
+                >>> xid_queries = [
```
Comment (sighol) [inline cognite/client/_api/datapoints.py:1291]: super optional nit: It took me a second to connect that `xid` means `external_id` in this context. Maybe rename to `external_id_queries`?

Comment (haakonvt) [reply cognite/client/_api/datapoints.py:1291]: Sure, as an example I think clarity is essential 👌

**Review by sighol — APPROVED** [review]
LGTM!

**Pull request merged**